tests/TestRegistryTest.cpp

```diff
@@ -323,3 +323,38 @@ TEST(TestRegistry, ResetPluginsWorks)
     myRegistry->resetPlugins();
     LONGS_EQUAL(0, myRegistry->countPlugins());
 }
+
+TEST(TestRegistry, listGroupNames)
+{
+    test1->setGroupName("GROUP_1");
+    myRegistry->addTest(test1);
+    test2->setGroupName("GROUP_2");
+    myRegistry->addTest(test2);
+    test3->setGroupName("GROUP_1");
+    myRegistry->addTest(test3);
+
+    myRegistry->listTestGroupNames(*result);
+    SimpleString s = output->getOutput();
+    STRCMP_CONTAINS("GROUP_1 ", s.asCharString());
+    STRCMP_CONTAINS("GROUP_2 ", s.asCharString());
+    LONGS_EQUAL(1, s.count("GROUP_1 "));
```

Comment:
Because the order of the groups and tests is not very important, in fact, the tests are listed backward (so the tests are executed from the most advanced to the simplest, defined at the file beginning, which was confusing me initially).
I found this approach more universal. 

If you think different, point me the best solution. 
And BTW I hope it will be merged soon into the trunk, otherwise we can't use such a unofficial modification.

Regards, 
MM